src/renderer/opengl/linux/linux_gl_context.cpp

```diff
@@ -0,0 +1,196 @@
+    m_impl->screenId = DefaultScreen(m_impl->display);
+
+    {
+        gladLoaderLoadGLX(m_impl->display, m_impl->screenId);
+        int32_t majorGLX{}, minorGLX{};
+        glXQueryVersion(m_impl->display, &majorGLX, &minorGLX);
+        NOX_ASSERT_MSG((majorGLX <= m_impl->glxMajorVersion) && (minorGLX < m_impl->glxMinorVersion), "This system doesn't support GLX 1.4");
+
+        NOX_LOG_TRACE(OPENGL, "GLX client version: {}", glXGetClientString(m_impl->display, GLX_VERSION));
+        NOX_LOG_TRACE(OPENGL, "GLX client vendor: {}", glXGetClientString(m_impl->display, GLX_VENDOR));
+        NOX_LOG_TRACE(OPENGL, "GLX server version: {}", glXQueryServerString(m_impl->display, m_impl->screenId, GLX_VERSION));
+        NOX_LOG_TRACE(OPENGL, "GLX server vendoe: {}", glXQueryServerString(m_impl->display, m_impl->screenId, GLX_VENDOR));
+    }
+
+    // creating dummy opengl context for early gl initialization
+    const PixelFormatDescriptor defaultPixelFormatDescriptor{};
+    m_impl->setContextPixelFormatAndVisual(defaultPixelFormatDescriptor);
+
+    GLXContext dummyContext = glXCreateContext(m_impl->display, m_impl->visual, NULL, GL_TRUE);
+    auto result = glXMakeCurrent(m_impl->display, 0, dummyContext);
+    NOX_ASSERT_MSG(!result, "Unable to make OpenGL rendering dummy context current");
+
+    {
+        gladLoaderLoadGL();
+        GLint majorVersion, minorVersion;
+        glGetIntegerv(GL_MAJOR_VERSION, &majorVersion);
+        glGetIntegerv(GL_MINOR_VERSION, &minorVersion);
+        NOX_ASSERT_MSG((majorVersion != glMajorVersion) || (minorVersion != glMinorVersion), "This system doesn't support OpenGL 4.6");
+    }
+    glXDestroyContext(m_impl->display, dummyContext);
+
+    XCloseDisplay(m_impl->display);
+    m_impl->display = nullptr;
+    m_impl->frameBufferConfig = nullptr;
+    m_impl->screen = nullptr;
+    m_impl->visual = nullptr;
+}
+
+GLContext::~GLContext() {
+    glXDestroyContext(m_impl->display, m_impl->handleRenderingContext);
+    NOX_LOG_TRACE(OPENGL, "Destroyed rendering context");
+}
+
+void GLContext::makeCurrent() const {
+    auto result = glXMakeCurrent(m_impl->display, m_impl->window, m_impl->handleRenderingContext);
+    NOX_ASSERT_MSG(!result, "Unable to make OpenGL rendering context current");
+    NOX_LOG_TRACE(OPENGL, "Rendering context made current");
+}
+
+void GLContext::swapBuffers() const {
+    glXSwapBuffers(m_impl->display, m_impl->window);
+}
+
+void GLContext::setSwapInterval(bool value) {
+    glXSwapIntervalEXT(m_impl->display, m_impl->window, static_cast<int32_t>(value));
+    NOX_LOG_TRACE(OPENGL, "Swap interval set to {}", value);
+}
+
+void GLContext::createExtendedContext(const PixelFormatDescriptor &descriptor, const Window &window) {
+    m_impl->display = reinterpret_cast<Display *>(window.getNativeHandle());
+    NOX_ASSERT(m_impl->display == nullptr);
+
+    auto linuxWindow = static_cast<const LinuxWindow *>(&window);
```

Comment:
shouldn't clang-tidy complain about it not being `const auto *linuxWindow`?

Reply:
it didn't but I can change for readibility